### PR TITLE
Fix: Add period to cloudfare record value

### PR DESCRIPTION
**Pull request by isaschm**
Addresses #26 

From what I have gathered, this should fix the issue. I'm not entirely sure how to test that though 🤔 
Do I need to run it to see if a new record is created or can I somehow gather that from the checks?

### CODE REVIEW

**Review by jgraeger — CHANGES_REQUESTED** [review]
I agree, testing this is hard. I extracted all relevant parts into a separate, minimal but sufficient terraform project (only `google_dns_managed_zone, google_dns_record_set, cloudflare_record`) and tested with a seperate GCP project an domain linked to cloudflare.

Here is the output of terraform plan:
```
~ resource "cloudflare_record" "test_graeger_dev" {
        id              = "4f1cbc93275a75476494c236f038ef85"
        name            = "tftest"
      ~ value           = "ns-cloud-d4.googledomains.com" -> "ns-cloud-d4.googledomains.com.."
        # (10 unchanged attributes hidden)
    }
```

After reverting the change, the output was:
```
~ resource "cloudflare_record" "test_graeger_dev" {
        id              = "4f1cbc93275a75476494c236f038ef85"
        name            = "tftest"
      ~ value           = "ns-cloud-d4.googledomains.com" -> "ns-cloud-d4.googledomains.com."
        # (10 unchanged attributes hidden)
    }
```

So it seems like that the google provider outputs a name servers domain name with a trailing dot, which gets dropped by the cloudflare provider or api. Therefore, i guess this fix will unfortunately not work 😕.

[I found a discussion on this in the cloudflare terraform project](https://github.com/cloudflare/terraform-provider-cloudflare/issues/154), providing multiple ideas for solving this.

I guess, we could use terraforms [replace](https://www.terraform.io/language/functions/replace) without any harm, because the DNS names will be canonically the same in the end. What do you think @isaschm 
@ldb?

**Comment by isaschm** [comment]
Wow, funky! 
I committed a solution using `replace` (the forward slashes are so `replace` treats the value as a regex). Again without much way to test but a regex validator 🙈 In the discussion they proposed using SuppressDiff. In case this record value doesn't change and implementing it doesn't take too much time, maybe that's also a solution?

File: dns.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@ resource "cloudflare_record" "live_mindtastic_lol" {
 
   zone_id = var.cloudflare_zone_id
   name    = "live"
-  value   = google_dns_managed_zone.live.name_servers[count.index]
+  value   = replace(google_dns_managed_zone.live.name_servers[count.index], "/[.]$/", "")
```
Comment (jgraeger) [inline dns.tf:18]:
```suggestion
  value   = trimsuffix(google_dns_managed_zone.live.name_servers[count.index], ".")
```

**Review by jgraeger — APPROVED** [review]
Thank you @isaschm!

I've tested your solution with the described setup and it seems to work as intended.
While working on different infrastructure stuff I stumbled over the [`trimsuffix`](https://www.terraform.io/language/functions/trimsuffix) function in the terraform language. That should do the same. I've added a suggestion, go with what you prefer more.

The `DiffSuppressFunc` discussion in the issue seems to be related to the provider code itself, not something we should touch here :)

Again, thank you for investigating this. Looks good from my side 👍

File: dns.tf (hunk 1 above)
Comment (isaschm) [reply dns.tf]: Uhhh that’s a lot cleaner!!

**Review by ldb — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**